modules/phase_field/src/userobjects/PolycrystalVoronoi.C

```diff
@@ -166,6 +243,59 @@ PolycrystalVoronoi::precomputeGrainStructure()
         _centerpoints[grain](2) = _bottom_left(2) + _range(2) * 0.5;
     }
   }
+
+  // Build a KDTree that is used to speedup point search
+  buildSearchTree();
```

Comment:
Does building the tree take a long time? If so, should we only do this if `use_kdtree` is set to true?

Reply:
```
PolycrystalVoronoi::buildSearchTree()
{
  if (!_use_kdtree)
    return;
```